### ISSUE TITLE
Cannot get result() since the metric has not yet been built

### ISSUE THREAD

**Issue by Shikamaru-Nara1729** [keras-team/keras#20188]
Basically the heading. I am attaching the colab notebook for reproducibility purpose, and have printed out the tensorflow and keras versions for perusal. [Link here](https://colab.research.google.com/drive/1F5Wzq6AqbYTzlBZHXypgiTn9xUVMOhyt#scrollTo=PUKQ6_Vv3nGk)

I think I'm doing everything the standard way. Cannot figure out why this error occurs. Of course this is a simpler code snippet than the one I'm working on which cannot be shared for obvious purposes, but this code demonstrates the issue.

Any clue on how to fix this/circumvent, or which version of keras/tf can i switch to, to avoid this issue?

If it is any help, I saw another [issue](https://github.com/keras-team/keras-nlp/issues/1716) open on keras-nlp, which discusses the same problem, so i'm guessing it's an issue with CompileMetrics module in the trainers folder, which is where the error points anyway.

**Comment by rohitcodism** [comment]
What is the fix of this error cause I am also getting the same error ? Can anyone help me resolving it.

**Comment by ghsanti** [comment]
> What is the fix of this error cause I am also getting the same error ? Can anyone help me resolving it.

If you can open a new issue w versions used and a minimal code that reproduces the error, and it will likely get fixed.
